src/anlz.rs

```diff
@@ -24,87 +24,83 @@
 //! - <https://djl-analysis.deepsymmetry.org/rekordbox-export-analysis/anlz.html>
 //! - <https://reverseengineering.stackexchange.com/questions/4311/help-reversing-a-edb-database-file-for-pioneers-rekordbox-software>
 
-use crate::util::{nom_input_error_with_kind, ColorIndex};
-use nom::error::ErrorKind;
-use nom::IResult;
+#![allow(clippy::must_use_candidate)]
+
+use crate::util::ColorIndex;
+use binrw::{
+    binrw,
+    io::{Read, Seek},
+    BinRead, BinResult, BinWrite, NullWideString, ReadOptions,
+};
+use modular_bitfield::prelude::*;
 
-#[derive(Debug)]
 /// The kind of section.
+#[derive(Debug, PartialEq, Clone)]
+#[binrw]
+#[brw(big)]
 pub enum ContentKind {
     /// File section that contains all other sections.
+    #[brw(magic = b"PMAI")]
     File,
     /// All beats found in the track.
+    #[brw(magic = b"PQTZ")]
     BeatGrid,
     /// Either memory points and loops or hotcues and hot loops of the track.
     ///
     /// *Note:* Since the release of the Nexus 2 series, there also exists the `ExtendedCueList`
     /// section which can carry additional information.
+    #[brw(magic = b"PCOB")]
     CueList,
     /// Extended version of the `CueList` section (since Nexus 2 series).
+    #[brw(magic = b"PCO2")]
     ExtendedCueList,
     /// Single cue entry inside a `ExtendedCueList` section.
+    #[brw(magic = b"PCP2")]
     ExtendedCue,
     /// Single cue entry inside a `CueList` section.
+    #[brw(magic = b"PCPT")]
     Cue,
     /// File path of the audio file.
+    #[brw(magic = b"PPTH")]
     Path,
     /// Seek information for variable bitrate files.
+    #[brw(magic = b"PVBR")]
     VBR,
     /// Fixed-width monochrome preview of the track waveform.
+    #[brw(magic = b"PWAV")]
     WaveformPreview,
     /// Smaller version of the fixed-width monochrome preview of the track waveform (for the
     /// CDJ-900).
+    #[brw(magic = b"PWV2")]
     TinyWaveformPreview,
     /// Variable-width large monochrome version of the track waveform.
     ///
     /// Used in `.EXT` files.
+    #[brw(magic = b"PWV3")]
     WaveformDetail,
     /// Fixed-width colored version of the track waveform.
     ///
     /// Used in `.EXT` files.
+    #[brw(magic = b"PWV4")]
     WaveformColorPreview,
     /// Variable-width large colored version of the track waveform.
     ///
     /// Used in `.EXT` files.
+    #[brw(magic = b"PWV5")]
     WaveformColorDetail,
     /// Describes the structure of a sond (Intro, Chrous, Verse, etc.).
     ///
     /// Used in `.EXT` files.
+    #[brw(magic = b"PSSI")]
     SongStructure,
     /// Unknown Kind.
     Unknown([u8; 4]),
 }
```

Comment:
Could you add a short explanation for each Unknown why that couldn't be removed?

Reply:
https://github.com/Holzhaus/rekordcrate/pull/47/commits/04b987f73ce5c4cb3bccd63cf38dd67806cf9c64

Reply:
Without this, we wouldn't be able to parse `*.2EX` files. See #9.